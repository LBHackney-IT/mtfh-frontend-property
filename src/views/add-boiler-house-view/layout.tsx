import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { AddBoilerHouseForm } from "../../components/add-boiler-house-form";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

interface Props {
  asset: Asset;
}

export const AddBoilerHouseLayout = ({  asset }: Props) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [requestError, setRequestError] = useState<string | null>(null);

  return (
    <>
      <Link as={RouterLink} to={`/property/${asset.id}`} variant="back-link">
        Back to asset
      </Link>

      {showSuccess && (
        <StatusBox
          variant="success"
          title={
            <span>
              The boiler house has been added successfully.{" "}
              <Link as={RouterLink} to={`/property/${asset.id}`}>
                Return to property
              </Link>
            </span>
          }
        />
      )}

      <h1 className="lbh-heading-h1">Add boiler house to property</h1>
      <span className="govuk-caption-m lbh-caption">Some subtitle text</span>

      {requestError && (
        <ErrorSummary
          id="patch-asset-error"
          title="Unexpected error"
          description={requestError || undefined}
        />
      )}

      <AddBoilerHouseForm
        setShowSuccess={setShowSuccess}
        setRequestError={setRequestError}
        asset={asset}
      />
    </>
  );
};
