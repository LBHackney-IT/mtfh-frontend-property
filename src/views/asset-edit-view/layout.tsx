import React, { useEffect, useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { locale } from "../../services";

import { Address, getAddressViaUprn } from "@mtfh/common/lib/api/address/v1";
import { ErrorSummary, Link } from "@mtfh/common/lib/components";
import { EditableAddress } from "../../components/edit-asset-address-form/editable-address";
import { ReferenceAddress } from "../../components/edit-asset-address-form/reference-address";

import "./styles.scss";



export interface AssetEditLayoutProperties {
    assetDetails: Asset;
}


export const AssetEditLayout = ({ assetDetails }: AssetEditLayoutProperties): JSX.Element => {

    const [llpgAddress, setLlpgAddress] = useState<Address | null>(null);
    const [showError, setShowError] = useState<boolean>(false);
    const [errorMessage, setErrorMessage] = useState<string | null>(null);


    useEffect(() => {
        getAddressViaUprn(assetDetails.assetAddress.uprn).then((searchAddressResponse) => {
            if (searchAddressResponse.addresses) {
                setLlpgAddress(searchAddressResponse.addresses[0])
            }
        })
    }, []);

    return (
        <>
            <Link as={RouterLink} to={`/property/${assetDetails.id}`} variant="back-link">
                {locale.asset}
            </Link>
            <h1 className="lbh-heading-h1">Edit property address</h1>
            <span className="govuk-caption-m lbh-caption">New Addresses are suggested from the Local Gazetteer to bring some standardisation.</span>
            
            {showError && 
                  <ErrorSummary
                  id="patch-asset-error"
                  title={locale.errors.unableToPatchAsset}
                  description={errorMessage ? errorMessage : undefined}
                />
            }
            
            <div className="mtfh-address-details">
                <section><EditableAddress llpgAddress={llpgAddress} assetDetails={assetDetails} setShowError={setShowError} setErrorMessage={setErrorMessage}/></section>
                <section><ReferenceAddress assetAddressDetails={assetDetails.assetAddress} /></section>
            </div>
        </>
    );
};
