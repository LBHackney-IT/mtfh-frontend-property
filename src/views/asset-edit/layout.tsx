import React, { FC } from "react";
import { Link as RouterLink } from "react-router-dom";
import { Asset } from "@mtfh/common/lib/api/asset/v1";
import {
    Alert as AlertIcon,
    Button,
    Center,
    CommentList,
    Heading,
    Layout,
    Link,
    PageAnnouncement,
    PageAnnouncementProvider,
    SideBar,
    SideBarProps,
    Spinner,
    WorkOrderList,
} from "@mtfh/common/lib/components";


export interface AssetEditLayoutProperties {
    assetDetails: Asset;
}

export const AssetEditLayout: FC<AssetEditLayoutProperties> = ({ assetDetails }) => {


    return (
        <>
            <h3>Asset Edit View</h3>
            <p>{JSON.stringify(assetDetails)}</p>
        </>
        //   <PageAnnouncementProvider sessionKey="asset">
        //     <PageAnnouncement />
        //     <Layout
        //       backLink={
        //         <Link as={RouterLink} to="/search" variant="back-link">
        //           {locale.backToSearch}
        //         </Link>
        //       }
        //       top={
        //         <Heading variant="h1">
        //           {alertsData.alerts?.length > 0 && (
        //             <AlertIcon
        //               viewBox="0 0 37 58"
        //               width="28"
        //               height="44"
        //               style={{ margin: "-2px 4px 0 0" }}
        //             />
        //           )}
        //           {locale.assetDetails.address(assetDetails.assetAddress)}
        //         </Heading>
        //       }
        //       side={<AssetSideBar assetDetails={assetDetails} alerts={alertsData.alerts} />}
        //     >
        //       <PropertyBody assetId={assetDetails.assetId} propertyId={assetDetails.id} />
        //     </Layout>
        //   </PageAnnouncementProvider>
    );
};