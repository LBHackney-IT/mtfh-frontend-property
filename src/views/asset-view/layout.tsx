import React, { FC } from "react";
import { Link as RouterLink } from "react-router-dom";

import { AssetSideBar } from "../../components/asset-sidebar";
import { PropertyBody } from "../../components/property-body";
import { locale } from "../../services";
import { useCautionaryAlerts } from "./utils";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import {
  Alert as AlertIcon,
  Button,
  Center,
  Heading,
  Layout,
  Link,
  PageAnnouncement,
  PageAnnouncementProvider,
  Spinner,
} from "@mtfh/common/lib/components";

import "./styles.scss";

export interface Props {
  assetDetails: Asset;
  assetChildren: Asset[] | undefined;
  showTenureInformation: boolean;
  showCautionaryAlerts: boolean;
  enableNewProcesses: boolean;
}

export const AssetLayout: FC<Props> = ({
  assetDetails,
  assetChildren,
  showTenureInformation,
  showCautionaryAlerts,
  enableNewProcesses,
}) => {
  const { alertsData, isLoading: isLoadingCautionaryAlerts } = useCautionaryAlerts(
    assetDetails.assetId,
    showCautionaryAlerts,
  );

  const tenure = useTenure(assetDetails.tenure ? assetDetails.tenure.id : null).data;

  if (assetDetails.tenure && tenure && alertsData.length > 0) {
    const { householdMembers } = tenure;

    alertsData.forEach((alert) => {
      const matchingMember = householdMembers.find(
        (householdMember) =>
          alert.personName && alert.personName === householdMember.fullName,
      );

      if (matchingMember) {
        alert.personId = matchingMember.id;
      }
    });

  }


  if (isLoadingCautionaryAlerts) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  return (
    <PageAnnouncementProvider sessionKey="asset">
      <PageAnnouncement />
      <Layout
        backLink={
          <Link as={RouterLink} to="/search" variant="back-link">
            {locale.backToSearch}
          </Link>
        }
        top={
          <Heading variant="h1">
            {showCautionaryAlerts && alertsData.length > 0 && (
              <span data-test="cautionary-alerts-icon">
                <AlertIcon
                  viewBox="0 0 37 58"
                  width="28"
                  height="44"
                  style={{ margin: "-2px 4px 0 0" }}
                />
              </span>
            )}
            {locale.assets.assetDetails.address(assetDetails.assetAddress)}
          </Heading>
        }
        side={
          <AssetSideBar
            assetDetails={assetDetails}
            alerts={alertsData}
            showTenureInformation={showTenureInformation}
            showCautionaryAlerts={showCautionaryAlerts}
          />
        }
      >
        <PropertyBody
          assetDetails={assetDetails}
          childAssets={assetChildren}
          enableNewProcesses={enableNewProcesses}
          id={assetDetails.id}
        />
        
      </Layout>
    </PageAnnouncementProvider>
  );
};
