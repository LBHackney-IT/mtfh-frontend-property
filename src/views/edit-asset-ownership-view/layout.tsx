import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { EditAssetOwnershipForm } from "../../components/edit-asset-ownership-form";
import { locale } from "../../services";

import { Asset } from "@mtfh/common/lib/api/asset/v1";
import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

interface EditAssetOwnershipLayoutProps {
  asset: Asset;
}

export const EditAssetOwnershipLayout = ({ asset }: EditAssetOwnershipLayoutProps) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [requestError, setRequestError] = useState<string | null>(null);

  return (
    <>
      <Link as={RouterLink} to={`/property/${asset.id}`} variant="back-link">
        Back to asset
      </Link>

      {showSuccess && (
        <StatusBox
          variant="success"
          data-testid="ownership-edit-success"
          // @ts-ignore No overload matches this call
          title={
            <span>
              {locale.lbhOwnershipInformation.editLbhOwnershipSuccessMessage}.{" "}
              <Link as={RouterLink} to={`/property/${asset.id}`}>
                View property
              </Link>
            </span>
          }
        />
      )}

      <h1 className="lbh-heading-h1">Edit asset ownership</h1>

      {requestError && (
        <ErrorSummary
          id="patch-asset-error"
          data-testid="ownership-edit-failure"
          title={locale.lbhOwnershipInformation.editOwnershipPatchError}
          description={locale.errors.tryAgainOrContactSupport}
        />
      )}

      <EditAssetOwnershipForm
        setShowSuccess={setShowSuccess}
        setRequestError={setRequestError}
        asset={asset}
      />
    </>
  );
};
