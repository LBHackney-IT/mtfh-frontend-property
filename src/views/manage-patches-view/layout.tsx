import React, { useState } from "react";
import { BrowserRouter, Link as RouterLink } from "react-router-dom";

import { PatchAssignmentForm } from "../../components/patch-assignment-form";

import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

export const ManagePatchesLayout = ({ assetId }: { assetId: string | undefined }) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const [requestError, setRequestError] = useState<string | null>(null);

  const backLink = assetId ? `/property/${assetId}` : "/search";
  const backLinkText = assetId ? "Back to property" : "Back to search";

  return (
    <>
      <BrowserRouter>
        <Link as={RouterLink} to={backLink} variant="back-link">
          {backLinkText}
        </Link>
      </BrowserRouter>
      {showSuccess && (
        <StatusBox variant="success" title="The patches have been updated successfully">
          <BrowserRouter>
            <Link as={RouterLink} to={backLink} variant="back-link">
              {backLinkText}
            </Link>
          </BrowserRouter>
        </StatusBox>
      )}

      <h1 className="lbh-heading-h1">Patch reassignment</h1>

      {requestError && (
        <ErrorSummary
          id="patch-asset-error"
          title="Unexpected error"
          description={requestError || undefined}
        />
      )}

      <PatchAssignmentForm
        setShowSuccess={setShowSuccess}
        setRequestError={setRequestError}
      />
    </>
  );
};
