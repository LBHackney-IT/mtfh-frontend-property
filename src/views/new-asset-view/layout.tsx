import React, { useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { NewAsset } from "../../components/new-asset-form";
import { locale } from "../../services";

import { CreateNewAssetRequest } from "@mtfh/common/lib/api/asset/v1";
import { ErrorSummary, Link, StatusBox } from "@mtfh/common/lib/components";

export const NewPropertyLayout = (): JSX.Element => {
  const [newProperty, setNewProperty] = useState<CreateNewAssetRequest | null>();
  const [showSuccess, setShowSuccess] = useState<boolean>(false);
  const [showError, setShowError] = useState<boolean>(false);
  const [errorHeading, setErrorHeading] = useState<string | null>(null);
  const [errorDescription, setErrorDescription] = useState<string | null>(null);

  return (
    <>
      <Link as={RouterLink} to="#" variant="back-link">
        Back
      </Link>
      <h1 className="lbh-heading-h1">New property</h1>
      <span
        id="new-property-disclaimer"
        className="govuk-hint lbh-hint"
        data-testid="new-property-disclaimer"
      >
        A new property will immediately be available for users of Manage My Home.
        Information is validated in most fields, but all protections are not in place, so
        some selections may result in mismatched information. It is recommended to
        thoroughly check the submission before adding the new asset.{" "}
      </span>
      {showSuccess && (
        <StatusBox variant="success" title={locale.assets.newPropertyAddedSuccessMessage}>
          <div>
            {" "}
            <Link as={RouterLink} to={`/property/${newProperty?.id}`}>
              View property
            </Link>{" "}
          </div>
        </StatusBox>
      )}

      {showError && (
        <ErrorSummary
          id="patch-asset-error"
          title={errorHeading || ""}
          description={errorDescription || undefined}
        />
      )}

      <section>
        <NewAsset
          setShowError={setShowError}
          setErrorHeading={setErrorHeading}
          setErrorDescription={setErrorDescription}
          setShowSuccess={setShowSuccess}
          setNewProperty={setNewProperty}
        />
      </section>
    </>
  );
};
