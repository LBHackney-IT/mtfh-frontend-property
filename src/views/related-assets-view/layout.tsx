import React, { useEffect, useState } from "react";
import { Link as RouterLink } from "react-router-dom";

import { RelatedAssets } from "../../components/related-assets/related-assets";
import { RelatedAsset, getAllRelatedAssets, organiseRelatedAssetsByType } from "./utils";

import { Asset, ParentAsset } from "@mtfh/common/lib/api/asset/v1";
import { Center, Link, Spinner } from "@mtfh/common/lib/components";

import "./styles.scss";

interface RelatedAssetsLayoutProps {
  asset: Asset;
  parentAssets: ParentAsset[];
  childrenAssets: Asset[] | undefined;
  loading: boolean;
}

export const RelatedAssetsLayout = ({
  asset,
  parentAssets,
  childrenAssets,
  loading,
}: RelatedAssetsLayoutProps): JSX.Element => {
  const [relatedAssets, setRelatedAssets] = useState<RelatedAsset[]>([]);
  const [relatedAssetsByType, setRelatedAssetsByType] = useState<any>(undefined);

  useEffect(() => {
    if (parentAssets?.length || childrenAssets) {
      setRelatedAssets(getAllRelatedAssets(parentAssets, childrenAssets));
    }
  }, [parentAssets, childrenAssets]);

  useEffect(() => {
    if (relatedAssets.length) {
      const assetsByType = organiseRelatedAssetsByType(relatedAssets);
      setRelatedAssetsByType(assetsByType);
    }
  }, [relatedAssets]);

  const assetHasRelatedAssets = (): boolean => {
    return !!relatedAssets.length;
  };

  const renderRelatedAssets = () => {
    if (loading) {
      return (
        <Center>
          <Spinner />
        </Center>
      );
    }

    // If we have related assets and the relatedAssetsByType object we render RelatedAssets components for each Asset type
    if (assetHasRelatedAssets() && relatedAssetsByType) {
      return Object.keys(relatedAssetsByType).map((assetType) => {
        return (
          <section key={assetType}>
            <RelatedAssets
              assetType={assetType}
              relatedAssets={relatedAssetsByType[assetType]}
            />
            <hr style={{ borderTop: "1px solid #e7eaec" }} />
          </section>
        );
      });
    }
    return (
      <p className="lbh-body-m" data-testid="no-related-assets-message">
        There are no related assets for this property.
      </p>
    );
  };

  return (
    <>
      <Link as={RouterLink} to={`/property/${asset.id}`} variant="back-link">
        Back to asset view
      </Link>
      <h1 className="lbh-heading-h1" data-testid="related-assets-heading">
        Related assets
      </h1>
      <p className="lbh-body-m" data-testid="property-asset-type">
        {asset.assetType}
      </p>
      <h2 className="lbh-heading-h2 margin-top-10" data-testid="property-address">
        {asset.assetAddress.addressLine1} - {asset.assetAddress.postCode}
      </h2>
      <hr style={{ borderTop: "1px solid #e7eaec" }} />

      {renderRelatedAssets()}
    </>
  );
};
