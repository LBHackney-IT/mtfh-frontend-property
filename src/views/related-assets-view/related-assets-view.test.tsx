import React from "react";
import { render } from "@hackney/mtfh-test-utils";
import { RelatedAssetsLayout } from "./layout";
import { Asset } from "@mtfh/common/lib/api/asset/v1";

const asset: Asset = {
    "id": "a65918aa-e441-bc34-e49d-fa8671f768da",
    "assetId": "00074866",
    "assetType": "HighRiseBlock",
    "rootAsset": "14edf718-19ff-ff43-4679-f8ef404fa029",
    "parentAssetIds": "14edf718-19ff-ff43-4679-f8ef404fa029",
    "assetLocation": {
        "floorNo": "",
        "totalBlockFloors": 0,
        "parentAssets": [
            {
                "type": "Estate",
                "id": "14edf718-19ff-ff43-4679-f8ef404fa029",
                "name": "Frampton Park Estate Frampton Park Road"
            }
        ]
    },
    "assetAddress": {
        "uprn": "",
        "addressLine1": "1-93 Pitcairn House St Thomass Square",
        "addressLine2": "Hackney",
        "addressLine3": "London",
        "addressLine4": "",
        "postCode": "E9 6PT",
        "postPreamble": ""
    },
    "assetManagement": {
        "agent": "Hackney Homes",
        "areaOfficeName": "Homerton (1) Panel Area Team",
        "isCouncilProperty": false,
        "managingOrganisation": "London Borough of Hackney",
        "managingOrganisationId": "c01e3146-e630-c2cd-e709-18ef57bf3724",
        "owner": "LBH",
        "isTMOManaged": false,
    },
    "assetCharacteristics": {
        "numberOfSingleBeds": 1,
        "numberOfDoubleBeds": 1,
        "numberOfFloors": 2,
        "numberOfBedrooms": 0,
        "numberOfLifts": 1,
        "numberOfLivingRooms": 0,
        "windowType": "",
        "yearConstructed": "1960",
        "totalBlockFloors": 1,
        "heating": "",
        "propertyFactor": "",
        "architecturalType": ""

    },
    "patches": [
        {
            "id": "bd0a8e2b-c3b5-4628-aa33-8e7509d5eac6",
            "parentId": "8d4fb05d-3ff5-48b7-a17a-71fcb27a66a8",
            "name": "SN4",
            "patchType": "patch",
            "domain": "MMH",
            "responsibleEntities": [
                {
                    "id": "7fb98e5e-6543-4e3e-b654-ac06e0a3f2f9",
                    "name": "Fake_Lyla Fake_Kling",
                    "responsibleType": "HousingOfficer"
                }
            ]
        }
    ],
    "boilerHouseId": "",
    "rentGroup": null,
    "tenure": null,
    "versionNumber": undefined
}

const childrenAssets: Asset[] = [
    {
        "id": "13e7cf17-60a0-729d-c297-a4e76a16b6fa",
        "assetId": "00023449",
        "assetType": "Dwelling",
        "assetAddress": {
            "uprn": "100023022363",
            "addressLine1": "61 PITCAIRN HOUSE",
            "addressLine2": "ST THOMASS SQUARE",
            "addressLine3": "HACKNEY",
            "addressLine4": "LONDON",
            "postCode": "E9 6PU",
            "postPreamble": ""
        },
        "tenure": {
            "id": "650b9052-9fc6-37cf-cafe-dd93d64646b2",
            "paymentReference": "1923449802",
            "startOfTenureDate": "2015-12-07T00:00:00Z",
            "endOfTenureDate": "",
            "type": "Secure",
            "isActive": true
        },
        "rootAsset": "14edf718-19ff-ff43-4679-f8ef404fa029",
        "parentAssetIds": "a65918aa-e441-bc34-e49d-fa8671f768da",
        "assetCharacteristics": {
            "numberOfSingleBeds": 1,
            "numberOfDoubleBeds": 1,
            "numberOfFloors": 2,
            "numberOfBedrooms": 0,
            "numberOfLifts": 1,
            "numberOfLivingRooms": 0,
            "windowType": "",
            "yearConstructed": "1965",
            "totalBlockFloors": 1,
            "heating": "",
            "propertyFactor": "",
            "architecturalType": ""
    
        },
        "assetManagement": {
            "agent": "Hackney Homes",
            "areaOfficeName": "Homerton (1) Panel Area Team",
            "isCouncilProperty": true,
            "managingOrganisation": "London Borough of Hackney",
            "managingOrganisationId": "c01e3146-e630-c2cd-e709-18ef57bf3724",
            "owner": "LBH",
            "isTMOManaged": false,
        },
        "assetLocation": {
            "floorNo": "5",
            "totalBlockFloors": 6,
            "parentAssets": []
        },
        "boilerHouseId": "",
        "rentGroup": null,
        "versionNumber": undefined
    }
]

test("it renders the component", async () => {
    const { container } = render(
        <RelatedAssetsLayout
            asset={asset}
            parentAssets={asset.assetLocation.parentAssets}
            childrenAssets={childrenAssets}
        />,
    );
    expect(container).toMatchSnapshot();
});

test("it displays the heading of the page, a back button, and the property whose related assets are being displayed", async () => {
    const { container } = render(
        <RelatedAssetsLayout
            asset={asset}
            parentAssets={asset.assetLocation.parentAssets}
            childrenAssets={childrenAssets}
        />,
    );
    expect(container).toHaveTextContent("Related assets");
    expect(container).toHaveTextContent(asset.assetAddress.addressLine1);
    expect(container).toHaveTextContent("Back to asset view");
});