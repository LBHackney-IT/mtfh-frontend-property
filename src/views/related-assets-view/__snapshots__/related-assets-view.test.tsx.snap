// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the component 1`] = `
<div>
  <a
    class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
    href="/property/a65918aa-e441-bc34-e49d-fa8671f768da"
  >
    Back to asset view
  </a>
  <h1
    class="lbh-heading-h1"
    data-testid="related-assets-heading"
  >
    Related assets
  </h1>
  <p
    class="lbh-body-m"
    data-testid="property-asset-type"
  >
    HighRiseBlock
  </p>
  <h2
    class="lbh-heading-h2 margin-top-10"
    data-testid="property-address"
  >
    1-93 Pitcairn House St Thomass Square
     - 
    E9 6PT
  </h2>
  <hr
    style="border-top: 1px solid #e7eaec;"
  />
  <section>
    <p
      class="lbh-body-m"
    >
      Estates
    </p>
    <div
      class="mtfh-card-list"
      data-testid="related-asset-items-group"
    >
      <div
        class="mtfh-link-box"
        data-testid="related-asset"
      >
        <div
          class="mtfh-search-card"
        >
          <div
            class="mtfh-link-overlay"
          >
            <a
              class="govuk-link lbh-link lbh-link--text-colour lbh-link--no-visited-state mtfh-search-tenure__title"
              href="/property/14edf718-19ff-ff43-4679-f8ef404fa029"
            >
              Frampton Park Estate Frampton Park Road
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
  <hr
    style="border-top: 1px solid #e7eaec;"
  />
  <section>
    <p
      class="lbh-body-m"
    >
      Dwellings
    </p>
    <div
      class="mtfh-card-list"
      data-testid="related-asset-items-group"
    >
      <div
        class="mtfh-link-box"
        data-testid="related-asset"
      >
        <div
          class="mtfh-search-card"
        >
          <div
            class="mtfh-link-overlay"
          >
            <a
              class="govuk-link lbh-link lbh-link--text-colour lbh-link--no-visited-state mtfh-search-tenure__title"
              href="/property/13e7cf17-60a0-729d-c297-a4e76a16b6fa"
            >
              61 PITCAIRN HOUSE
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
  <hr
    style="border-top: 1px solid #e7eaec;"
  />
</div>
`;
